django/db/models/base.py

```diff
@@ -1063,12 +1063,16 @@ def _save_table(
         for a single table.
         """
         meta = cls._meta
-        non_pks = [f for f in meta.local_concrete_fields if not f.primary_key]
+        non_pks_non_generated = [
+            f
+            for f in meta.local_concrete_fields
+            if not f.primary_key and not f.generated
+        ]
```

Comment:
This doesn't have to done here as this is meant as a non-invasive backport but since `non_pks_non_generated` is only used for containment checks and doesn't need to preserve ordering it might be a good candidate to use a `set` instead of a `list`.

Reply:
It could also be cached up on `Options`, it doesn’t need regenerating on every `save()` call.